catalog/plugin/catalog-plugin-metacardingest-network/src/test/java/org/codice/ddf/catalog/plugin/metacard/MetacardIngestNetworkPluginTest.java

```diff
@@ -0,0 +1,187 @@
+
+    @Mock
+    private Map<String, String> mockParsedAttributes;
+
+    @Mock
+    private KeyValueParser mockParser;
+
+    @Mock
+    private MetacardServices mockMetacardServices;
+
+    @Mock
+    private AttributeFactory mockAttributeFactory;
+
+    @Mock
+    private MetacardCondition mockMetacardCondition;
+
+    private List<Metacard> metacards;
+
+    private List<String> newAttributes;
+
+    private MetacardIngestNetworkPlugin plugin;
+
+    @Before
+    public void setup() throws Exception {
+
+        metacards = new ArrayList<>();
+        newAttributes = new ArrayList<>();
+
+        plugin = new MetacardIngestNetworkPlugin(mockParser,
+                mockMetacardServices,
+                mockAttributeFactory,
+                mockMetacardCondition);
+        plugin.setNewAttributes(newAttributes);
+
+        when(mockCreateRequest.getMetacards()).thenReturn(metacards);
+        when(mockParser.parsePairsToMap(newAttributes)).thenReturn(mockParsedAttributes);
+    }
+
+    @Test(expected = StopProcessingException.class)
+    public void testClientInfoMapNull() throws Exception {
+        ThreadContext.put(CLIENT_INFO_KEY, null);
+        plugin.processPreCreate(mockCreateRequest);
+    }
+
+    @Test(expected = StopProcessingException.class)
+    public void testClientInfoMapNotMap() throws Exception {
+        ThreadContext.put(CLIENT_INFO_KEY, new Object());
+        plugin.processPreCreate(mockCreateRequest);
+    }
+
+    @Test
+    public void testApplySuceeds() throws Exception {
+        ThreadContext.put(CLIENT_INFO_KEY, INFO_MAP);
+        when(mockMetacardCondition.applies(INFO_MAP)).thenReturn(true);
+        plugin.processPreCreate(mockCreateRequest);
+        verify(mockParser).parsePairsToMap(newAttributes);
+        verify(mockMetacardServices).setAttributesIfAbsent(metacards,
+                mockParsedAttributes,
+                mockAttributeFactory);
+        verifyNoMoreInteractions(mockParser, mockMetacardServices);
+    }
+
+    @Test
+    public void testApplyFails() throws Exception {
+        ThreadContext.put(CLIENT_INFO_KEY, INFO_MAP);
+        when(mockMetacardCondition.applies(INFO_MAP)).thenReturn(false);
+        plugin.processPreCreate(mockCreateRequest);
+        verifyZeroInteractions(mockParser, mockMetacardServices);
+    }
+
+    @Test
+    public void testGettersAndSetters() throws Exception {
+        MetacardIngestNetworkPlugin networkPlugin = new MetacardIngestNetworkPlugin(mockParser,
+                mockMetacardServices);
+
+        networkPlugin.setCriteriaKey(CRITERIA_KEY);
+        networkPlugin.setExpectedValue(EXPECTED_VALUE);
+        networkPlugin.setNewAttributes(newAttributes);
+
+        assertThat(networkPlugin.getCriteriaKey(), is(CRITERIA_KEY));
+        assertThat(networkPlugin.getExpectedValue(), is(EXPECTED_VALUE));
+        assertThat(networkPlugin.getNewAttributes(), is(newAttributes));
+    }
+
+    @Test
+    public void testPassthroughMethods() throws Exception {
```

Comment:
Should we split this into individual tests to make it clear we cover all the methods?